core/src/main/java/de/danoeh/antennapod/core/feed/LocalFeedUpdater.java

```diff
@@ -0,0 +1,109 @@
+package de.danoeh.antennapod.core.feed;
+
+import android.content.Context;
+import android.media.MediaMetadataRetriever;
+import android.net.Uri;
+import android.webkit.MimeTypeMap;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+import de.danoeh.antennapod.core.storage.DBTasks;
+
+public class LocalFeedUpdater {
+
+    private static long getFileDuration(File f) {
+        MediaMetadataRetriever mediaMetadataRetriever = new MediaMetadataRetriever();
+        mediaMetadataRetriever.setDataSource(f.getAbsolutePath());
+        String durationStr = mediaMetadataRetriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION);
+        return Long.parseLong(durationStr);
+    }
+
+    /** Starts the import process. */
+    public static void startImport(Uri uri, Context context) {
+        File f = new File(uri.getPath());
+        if (!f.isDirectory()) {
+            throw new RuntimeException("invalid path");
+        } else {
+            startImportDirectory(uri, context);
+        }
+    }
+
+    private static void startImportDirectory(Uri uri, Context context) {
+        //create a feed object for this directory
+        File f = new File(uri.getPath());
+        String dirUrl = uri.toString();
+        Feed dirFeed = new Feed(dirUrl, null, "Local directory (" + dirUrl + ")");
```

Comment:
I think using the directory name instead of the full uri looks better. I would also remove the text `local directory`.

If it exists, it would be great to use `Folder.jpg`, `folder.jpg`, `Folder.png` or `folder.png` as feed icon. That files are commonly used to create icons for folders.